### PR TITLE
Upgrade graphiql to v5

### PR DESCRIPTION
**Pull request by berryr0**
## 📌 Description

Upgraded graphiql and graphiql react

---

## 🔍 Changes Made

* upgraded graphiql/react to v0.35.4
   * [Migration Guide](https://github.com/graphql/graphiql/blob/main/docs/migration/graphiql-5.0.0.md)
   * Main updates in: packages/graphiql-plugin-multipart-requests/src/context.tsx
* upgraded graphiql to v5 in example
   * monico editor and comments in variables!! 
   * [v5 release notes](https://github.com/graphql/graphiql/releases/tag/graphiql%405.0.0)
* added github workflow for pr checks
* added github workflow for publishing
* added pr template
* Added .npmrc to point to public registry

- [x] Brief bullet-point list of changes
- [x] Highlight **key files or logic updated**
- [x] Mention **any breaking changes** or impacts

---

## 🧪 Testing

- [x] Tested locally
- [x] Unit tests updated/added
- [x] CI pipeline passed (lint, tests)

Describe **how you tested the changes** and any edge cases considered.

---

## 📷 Screenshots (if UI-related)
image

---

## ✅ Checklist

Please confirm all items are complete before requesting a review:

- [x] I have run `lint` and fixed any issues
- [x] I have run all existing tests and they pass
- [x] I have added tests for new code (if needed)
- [x] I have documented my changes (if needed)
- [x] Linked relevant issues in the PR description

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -14,6 +14,7 @@
     "demo:record": "RECORD_DEMO=true npx playwright test --config ./tests --project=chromium ./integration/demo.spec.ts",
     "demo:convert": "find . -type f -name \"demo.webm\" -exec bash -c 'ffmpeg -y -i \"$0\" -ss 1 -vf \"scale=1000:-1:flags=lanczos,split[s0][s1];[s0]palettegen[p];[s1][p]paletteuse=dither=bayer:bayer_scale=4\" -c:v gif \"${0%.webm}.gif\" && rm \"$0\"' {} \\;",
     "precommit": "npm run lint && npm run test && npm run release",
+    "publish:npm": "lerna publish from-package",
```
Comment (berryr0) [inline package.json:17]: This is untested but it should publish only packages that have a version greater than what's publicly published to npm

https://lerna.js.org/docs/features/version-and-publish#from-package

File: example/package.json [hunk 2]
```diff
@@ -13,9 +13,11 @@
   },
   "type": "module",
   "dependencies": {
-    "@graphiql/toolkit": "^0.9.1",
-    "@workday/graphiql-plugin-multipart-requests": "^0.0.1",
-    "graphiql": "^3.2.1",
+    "@graphiql/toolkit": "^0.11.3",
+    "@graphiql/plugin-doc-explorer": "0.3.0",
```
Comment (berryr0) [inline example/package.json:17]: This is a new exciting feature, the doc explorer is now a separate plugin so you can easily swap it out with a custom one if you want!

File: example/src/index.tsx [hunk 3]
```diff
@@ -12,6 +12,7 @@ import {
   MultipartAttachmentsContext,
   MultipartAttachmentsProvider,
 } from '@workday/graphiql-plugin-multipart-requests';
+import 'graphiql/setup-workers/webpack';
```
Comment (berryr0) [inline example/src/index.tsx:15]: This is part of the migration guide, and is required for the new monaco editor to have web workers.

File: example/webpack.config.cjs [hunk 4]
```diff
@@ -38,13 +38,6 @@ module.exports = {
     ]
   },
   resolve: {
-    // make sure we get the same libraries even if we symlink in dependencies
```
Comment (berryr0) [inline example/webpack.config.cjs:41]: I removed this because it was causing errors when running dev for me. It was causing dev to keep looking at incorrect dependency after the version was upgraded.

File: packages/graphiql-plugin-multipart-requests/src/plugin.tsx [hunk 5]
```diff
@@ -16,15 +17,15 @@ const GRAPHIQL_PLUGIN: GraphiQLPlugin = {
   title: 'Multipart Attachments',
   icon: () => {
     const [attachments] = useContext(MultipartAttachmentsInternalContext);
-    const pluginContext = usePluginContext();
+    const visiblePlugin = useGraphiQL(state => state.visiblePlugin);
```
Comment (berryr0) [inline packages/graphiql-plugin-multipart-requests/src/plugin.tsx:20]: useGraphiQL is now responsible for the following things:
https://github.com/graphql/graphiql/blob/main/packages/graphiql-react/README.md#available-stores

File: .github/workflows/publish.yaml [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+        uses: actions/setup-node@v4
+        with:
+          node-version: '20.x' # Todo use our .nvmrc instead
+          registry-url: 'https://registry.npmjs.org'
+      - name: Install dependencies and build 🔧
+        run: npm run setup && npm run release
+      - name: Publish package on NPM 📦
+        run: npm run publish:npm
+        env:
+          NODE_AUTH_TOKEN: ${{ secrets.PUBLISH_TO_NPM_TOKEN }}
```
Comment (berryr0) [inline .github/workflows/publish.yaml:21]: I added this token to github secrets

File: package.json (hunk 1 above)
Comment (neeraj-aduri-wday) [reply package.json:17]: Are we going to find out after merge?

File: example/package.json (hunk 2 above)
Comment (neeraj-aduri-wday) [reply example/package.json:17]: Are we using any features from this here?

File: package.json (hunk 1 above)
Comment (berryr0) [reply package.json:17]: After this merges, I will create the github release, and see if this works. Otherwise, I will have to manually have to publish to npm. (and debug why this didn't work)

File: example/package.json (hunk 2 above)
Comment (berryr0) [reply example/package.json:17]: I think we will want to use a custom doc explorer in graph-api-explorer. (Part of the motivation for this upgrade!)

File: package.json (hunk 1 above)
Comment (neeraj-aduri-wday) [reply package.json:17]: Please let me know if the release gets updated as expected!

**Review by neeraj-aduri-wday — APPROVED** [review]
Nice work. Looks good to me!
